catalog.tests.bom

```diff
@@ -0,0 +1,164 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+brooklyn.catalog:
+  id: brooklyn-postgresql-node-tests
+  version: 0.10.0-SNAPSHOT
+  itemType: template
+  iconUrl: https://wiki.postgresql.org/images/a/a4/PostgreSQL_logo.3colors.svg
+  name: PostgreSQL Node tests
+  description: Test that PostgreSQL Node is running correctly
+  license_code: Apache-2.0
+  license_url: http://www.apache.org/licenses/LICENSE-2.0.txt
+  item:
+    services:
+    - type: brooklyn-postgresql-node
+      id: postgresqlnode
+      name: PostgreSQL Node
+      brooklyn.config:
+        datastore.creation.script.contents: |
+          CREATE DATABASE temporary;
+          USE temporary;
+          CREATE TABLE users (name VARCHAR(20), owner VARCHAR(20), creation_date DATE, updated_date DATE);
+          INSERT INTO users VALUES ('mysql', 'mysql', CURRENT_DATE, CURRENT_DATE)
+        postgresql.port: 6789
+    - type: org.apache.brooklyn.test.framework.TestCase
+      name: PostgreSQL Node tests
+      targetId: postgresqlnode
+      brooklyn.children:
+
+      # Sensors tests
+      - type: org.apache.brooklyn.test.framework.TestCase
+        name: Sensors Tests
+        brooklyn.children:
+        - type: org.apache.brooklyn.test.framework.TestSensor
+          name: Check PostgreSQL server is up
+          sensor: service.isUp
+          timeout: 10m
+          assert:
+          - equals: true
+        - type: org.apache.brooklyn.test.framework.TestSensor
+          name: Check PostgreSQL port is setup correctly
+          sensor: postgresql.port
+          timeout: 1m
+          assert:
+          - equals: 6789
```

Comment:
Can we check that this port is reachable from the AMP server? (not with executing any database stuff; just that there is a TCP response from the ip:port). I think that currently this test would pass, even if the Postgres endpoint was not reachable due to incorrect iptables or security group rules.

It looks like we don't have that in `org.apache.brooklyn.test.framework`, but would be good to add it (e.g. a `TestPortReachable`, which would call `Networking.isReachable` - or something like that)?